saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationFilter.java

```diff
@@ -103,4 +110,16 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
 		return getAuthenticationManager().authenticate(authentication);
 	}
 
+	/**
+	 * Use the given {@link Saml2AuthenticationRequestRepository} to remove the saved
+	 * authentication request.
+	 * @param authenticationRequestRepository the
+	 * {@link Saml2AuthenticationRequestRepository} to use
+	 * @since 5.6
+	 */
+	public void setAuthenticationRequestRepository(
+			Saml2AuthenticationRequestRepository<AbstractSaml2AuthenticationRequest> authenticationRequestRepository) {
+		this.authenticationRequestRepository = authenticationRequestRepository;
```

Comment:
Please check for null. `Assert.notNull` is handy here.

Reply:
This method also probably needs to pass the provided request repository down to the authentication converter, if the converter is of the correct type.